### PR TITLE
Spring storage remove resource manager provider

### PR DESCRIPTION
**Pull request by saragluna**
Take over the PR https://github.com/Azure/azure-sdk-for-java/pull/15837. 

This PR is going to 
- Add a Spring extension of the `azure-identity` library
- Upgrade the resource manager dependencies to track2
- Break down the whole resourcemanager provider into separate resource providers for each service.

### CODE REVIEW

File: sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java [hunk 1]
```diff
@@ -3,79 +3,89 @@
...
+//                                   .withLogLevel(HttpLogDetailLevel.BASIC)
+                                   .authenticate(credential, profile)
+                                   .withDefaultSubscription();
+    }
+//
+//    @VisibleForTesting
+//    protected Azure authenticateToAzure(RestClient restClient, String subscriptionId,
+//                                        AzureTokenCredentials credentials) {
+//        return Azure.authenticate(restClient, credentials.domain()).withSubscription(subscriptionId);
+//    }
```
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:59]: the code you have marked as comment here, I am not sure whether it's appropriate, should we delete?

**Review by backwind1233 — COMMENTED** [review]
Up

File: sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java (hunk 1 above)
Comment (saragluna) [reply sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:59]: Yes, we should delete them.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml [hunk 2]
```diff
@@ -1,90 +1,120 @@
...
-  </build>
+
+      <!-- mvn azure-webapp:config -->  
+      <plugin> 
+        <groupId>com.microsoft.azure</groupId>  
+        <artifactId>azure-webapp-maven-plugin</artifactId>  
+        <version>1.12.0</version>  
+        <configuration> 
+          <schemaVersion>V2</schemaVersion>
+          <resourceGroup>[your-resource-group]</resourceGroup>
```
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml:99]: This is an Azure maven plugin to create an App Service and deploy your application jar file to the service.

**Comment by saragluna** [comment]
TODO:
- Set authority host.
- Add configuration processor for `azure.credential.` prefixed properties.

**Review by chenrujun — APPROVED** [review]

File: sdk/spring/azure-identity-spring/src/test/java/com/azure/spring/identity/PrefixedSpringCredentialBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Test
+    public void testWithNoEnvironmentSet() {
+        final PrefixedSpringCredentialBuilder builder = new PrefixedSpringCredentialBuilder();
+        assertThrows(IllegalArgumentException.class, builder::build);
+    }
+
+    @Test
+    public void testWithNoPrefixSet() {
+        final PrefixedSpringCredentialBuilder builder = new PrefixedSpringCredentialBuilder();
```
Comment (jialindai) [inline sdk/spring/azure-identity-spring/src/test/java/com/azure/spring/identity/PrefixedSpringCredentialBuilderTest.java:28]: Need to set one environment here?

File: sdk/spring/azure-identity-spring/src/main/java/com/azure/spring/identity/DefaultSpringCredentialBuilder.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    public TokenCredential build() {
+        if (environment == null) {
+            throw new IllegalArgumentException("To build a spring credential the environment must be set");
+        }
+
+        List<TokenCredential> tokenCredentials = new ArrayList<>();
+
+        if (alternativePrefix != null) {
+            tokenCredentials.add(populateTokenCredential(alternativePrefix));
```
Comment (jialindai) [inline sdk/spring/azure-identity-spring/src/main/java/com/azure/spring/identity/DefaultSpringCredentialBuilder.java:51]: If alternativePrefix is not null, then we will get managed identity credential if there is no such prefixed configuration in Environment. The order should be:

- Check alternativePrefix client-id, client-secret
- Check alternativePrefix client-id, client-certificate-path
- If there is only alternativePrefix client-id, try managed identity with client-id
- Check default prefix client-id, client-secret
- Check default prefix client-id, client-certificate-path
- If there is only default prefix client-id, try managed identity with client-id
- Use managed identity with no argument
-

File: sdk/spring/azure-identity-spring/src/test/java/com/azure/spring/identity/PrefixedSpringCredentialBuilderTest.java (hunk 3 above)
Comment (saragluna) [reply sdk/spring/azure-identity-spring/src/test/java/com/azure/spring/identity/PrefixedSpringCredentialBuilderTest.java:28]: Ah yes.

File: sdk/spring/azure-identity-spring/src/main/java/com/azure/spring/identity/DefaultSpringCredentialBuilder.java (hunk 4 above)
Comment (saragluna) [reply sdk/spring/azure-identity-spring/src/main/java/com/azure/spring/identity/DefaultSpringCredentialBuilder.java]: Changed.

**Comment by saragluna** [comment]
I'll skip the check of `java - core - ci` since it kept failing because `azure-core-amqp` and `azure-communication-common`, check the build [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=667890&view=results). But since the build passed several days ago, so it should be okay.

**Pull request merged**